### ISSUE TITLE
[Q]How do we define the metric name in swep.config?

### ISSUE THREAD

**Issue by hjzhang1018** [wandb/client#2519]
Hi I'm a newbie in using wandb. When I looked into the tutorials, I'm a bit confused about this metric name definition.
For example here,
```
sweep_config = {
    "method": "bayes",  # grid, random
    "metric": {"name": "train_loss", "goal": "minimize"},
    "parameters": {
        "num_train_epochs": {"values": [2, 3, 5]},
        "learning_rate": {"min": 5e-5, "max": 4e-4},
    },
}
```
I saw some people use metric name "train_loss", some use "val_loss", some use "loss". So I wonder if there's a dictionary where we could select the name from? What's the difference when we pass the metric name "val_loss" and "loss"?
If I want to optimize my hyperparameters by calculating "AUC_ROC", what exact name should I give to this sweep config? "auroc" or "auc_roc"? Does it matter if there's a typo error? 

Sorry I really couldn’t find a standard instructions for this one. Because I don’t know how sweep understand the metric we want it to optimize by just passing a "name" to it ?

Thanks very much! I would appreciate any help!

**Comment by vanpelt** [comment]
Hey @hjzhang1018 the metric will need to be one that you're logging with wandb.  You [log metrics](https://docs.wandb.ai/ref/python/log) in wandb by calling `wandb.log({"my_metric": X})`.  Certain integrations will automatically log metrics for you.  You can see what metrics are being logged for a run by looking at the "Summary" section of your run from the "Overview" section.

![image](https://user-images.githubusercontent.com/17/128756417-82451330-14d0-4979-8953-61e3a50be5dc.png)

In the example above I've logged "average_call_time" and "calls" as summary metrics in my run.  I could use either of these in my sweep config if I were running one.  Does that make sense?

**Comment by hjzhang1018** [comment]
Thank you @vanpelt! I used this `wandb.log({"my_metric": X})`, but it showed that 

> "X" is not defined.

I'm not very clear when I can call this `wandb.log` function. Is it after all the trainings ? But I need to set up my metric name in sweep_id before the hyperparameter screening. Could you explain a bit more in details? Thank you very much!

**Comment by vanpelt** [comment]
@hjzhang1018 X was meant to mean a variable.  When you train an ML model you generally have a loop that measures the performance of the model by calculating a performance metric like accuracy.  You would replace the X in my example with the calculated metric you care about.

You can find examples in various frameworks here: https://GitHub.com/wandb/examples
